File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> n=new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            int a=target-num;
            if(n.containsKey(a)){
                return new int[] {n.get(a),i};
            }
            n.put(num,i);
        }
        return new int[] {};
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        valToIndex = {}
        for i in range(len(nums)):
            rem = target - nums[i]

            if rem in valToIndex:
                return [valToIndex[rem], i]
            
            valToIndex[nums[i]] = i

        return []
        